src/ucp/core/ucp_worker.c

```diff
@@ -821,9 +821,9 @@ static int ucp_worker_iface_find_better(ucp_worker_h worker,
 static ucs_status_t ucp_worker_select_best_ifaces(ucp_worker_h worker,
                                                   uint64_t *tl_bitmap_p)
 {
-    ucp_context_h context = worker->context;
-    uint64_t tl_bitmap    = 0;
-    ucp_rsc_index_t repl_ifaces[context->num_tls];
+    ucp_context_h context        = worker->context;
+    uint64_t tl_bitmap           = 0;
+    ucp_rsc_index_t *repl_ifaces = ucs_alloca(context->num_tls);
```

Comment:
i would use `context->num_tls * sizeof(ucp_rsc_index_t)`, to not assume it's 1 byte